test/sol6/kyberNetwork.js

```diff
@@ -310,57 +311,99 @@ contract('KyberNetwork', function(accounts) {
             tempMatchingEngine = await MatchingEngine.new(admin);
             mockReserve = await MockReserve.new();
             gasHelperAdd = accounts[9];
-        })
-        beforeEach("global setup", async function(){
+            maliciousStorage = await MaliciousStorage.new();
+            feeHandler = accounts[3];
+        });
+
+        beforeEach("global setup", async() => {
             tempStorage = await KyberStorage.new(admin);
             tempNetwork = await KyberNetwork.new(admin, tempStorage.address);
             await tempStorage.setNetworkContract(tempNetwork.address, {from: admin});
 
             await tempNetwork.addOperator(operator, {from: admin});
-            await tempMatchingEngine.setNetworkContract(tempNetwork.address, {from: admin});
-            await tempMatchingEngine.setKyberStorage(tempStorage.address, {from: admin});
-            await tempStorage.setFeeAccountedPerReserveType(true, true, true, false, true, true, {from: admin});
-            await tempStorage.setEntitledRebatePerReserveType(true, false, true, false, true, true, {from: admin});
-
-            //init feeHandler
-            KNC = await TestToken.new("kyber network crystal", "KNC", 18);
-            proxyForFeeHandler = tempNetwork;
-            feeHandler = await FeeHandler.new(DAO.address, proxyForFeeHandler.address, tempNetwork.address, KNC.address, burnBlockInterval, DAO.address);
-            rateHelper = await RateHelper.new(admin);
-            await rateHelper.setContracts(tempMatchingEngine.address, DAO.address, tempStorage.address, {from: admin});
-
             await tempNetwork.setDAOContract(DAO.address, {from: admin});
         });
 
-        it("set empty fee handler contract", async function(){
+        it("set empty fee handler contract", async() => {
             await expectRevert(
                 tempNetwork.setContracts(zeroAddress, tempMatchingEngine.address, gasHelperAdd, {from: admin}),
                 "feeHandler 0"
             );
         });
 
-        it("set empty matching engine contract", async function(){
+        it("set empty matching engine contract", async() => {
             await expectRevert(
-                tempNetwork.setContracts(feeHandler.address, zeroAddress, gasHelperAdd, {from: admin}),
+                tempNetwork.setContracts(feeHandler, zeroAddress, gasHelperAdd, {from: admin}),
                 "matchingEngine 0"
             );
         });
 
-        it("should enable setting an empty dao contract", async function(){
+        it("should enable setting an empty dao contract", async() => {
             await tempNetwork.setDAOContract(zeroAddress, {from: admin});
             
             let rxContracts = await tempNetwork.getContracts();
             assert.equal(rxContracts.daoAddress, zeroAddress);
         });
+
+        it("should do nothing if setting to same DAO address", async() => {
+            await tempNetwork.setDAOContract(DAO.address, {from: admin});
```

Comment:
should check there is no event of updating DAO is emitted?

Reply:
OK, done.